core/src/main/java/org/apache/iceberg/InheritableMetadataFactory.java

```diff
@@ -50,13 +52,19 @@ static class BaseInheritableMetadata implements InheritableMetadata {
     private final long snapshotId;
     private final long sequenceNumber;
     private final String manifestLocation;
+    private Long nextRowId;
```

Comment:
I haven't thought through all the usages but should we have this be atomic? I think we may need handle multiple threads using this class at the same time.

Reply:
No, we create a separate `InheritableMetadata` for each manifest reader and the readers make no guarantees about thread safety.